evalml/automl/utils.py

```diff
@@ -72,14 +72,19 @@ def make_data_splitter(X, y, problem_type, problem_configuration=None, n_splits=
                                max_delay=problem_configuration.get('max_delay'))
 
 
-def tune_binary_threshold(pipeline, objective, problem_type, X_threshold_tuning, y_threshold_tuning):
+def tune_binary_threshold(pipeline, objective, problem_type, X_threshold_tuning, y_threshold_tuning, best_pipeline=False):
     """Tunes the threshold of a binary pipeline to the X and y thresholding data
 
     Arguments:
-        pipeline (Pipeline): Pipeline instance to threshold
-        X_threshold_tuning (ww.DataTable): Features to tune pipeline to
-        y_threshold_tuning (ww.DataColumn): Target data to tune pipeline to
+        pipeline (Pipeline): Pipeline instance to threshold.
+        objective (ObjectiveBase): The objective we want to tune with. If not tuneable and best_pipeline is True, will use to F1.
```

Comment:
@bchen1116 Can you explain this design choice? It feels arbitrary that we optimize the threshold against F1 when it's possible the user deliberately did not specify F1 as the primary objective or include it in `additional_objectives`.

Reply:
This was something @dsherry and I decided on. It seemed to be the best default objective to use since it gives a fair interpretation of the data for both balanced and imbalanced datasets. 

This is definitely worth looking into and discussing, if you have ideas.

Reply:
Why does the threshold tuning always need to happen even when the primary objective does not support it? 

My preference would be that the threshold would only be tuned for the best pipeline when the primary objective supports it for two reasons:

1. This is what's happening during search and I think it's important to keep the "training steps" the same between the best pipeline training and search
2. As a user, it feels weird that my pipeline is being tuned against something I did not specify as a search parameter.

Reply:
This is a good conversation.  The only thing I have to add is "True, will use F1".  Typo.  

With respect to the F1 point, I agree that its strange that objective has a kind of defacto default while it's required to be specified.  It seems that if you're trying to tune a threshold and the objective is not tunable...well, shoot, that sounds like an _exceptional_ situation, no? ;)

Reply:
Yep, great points. We can discuss this after standup today and figure out the best solution to go with!

Reply:
After discussion with @dsherry @freddyaboulton @angela97lin @chukarsten, we decided to remove this change to default to F1. We will leave `tune_threshold=True`, but we will not tune if the objective passed in isn't tunable.